components/ipfs/ipfs_service.cc

```diff
@@ -881,6 +971,100 @@ void IpfsService::OnPreWarmComplete(
     std::move(prewarm_callback_for_testing_).Run();
 }
 
+//{
+//  "PinLsList": {
+//    "Keys": {
+//      "<string>": {
+//        "Type": "<string>"
+//      }
+//    }
+//  },
+//  "PinLsObject": {
+//    "Cid": "<string>",
+//    "Type": "<string>"
+//  }
+//}
+void IpfsService::OnGetPinsResult(
+    APIRequestList::iterator iter,
+    GetPinsCallback callback,
+    api_request_helper::APIRequestResult response) {
+  int response_code = response.response_code();
+  requests_list_.erase(iter);
+
+  bool success = response.Is2XXResponseCode();
+
+  if (!success) {
+    VLOG(1) << "Fail to get pins, response_code = " << response_code;
+    std::move(callback).Run(false, absl::nullopt);
+    return;
+  }
+
+  ipfs::GetPinsResult result;
+  bool parse_result =
+      IPFSJSONParser::GetGetPinsResultFromJSON(response.body(), &result);
```

Comment:
Please use response.value_body() (returns base::Value) after rebasing. https://github.com/brave/brave-core/pull/15933 just merged.

Reply:
same for methods below